### ISSUE TITLE
Cannot import web-sdk

### ISSUE THREAD

**Issue by mk-ryan1988** [mParticle/mparticle-web-sdk#641]
![image](https://github.com/mParticle/mparticle-web-sdk/assets/35534717/f1429483-33eb-46de-9615-461a3cebf1c3)

Project details:
node: 14.15.1
vue: 2.x

(let me know if there's any other details you need)

Im aware thats not much to go on but I'm hoping you may have some context that may assist me getting this setup.
Cannot import the sdk (version 2.22.1) It's defiantly the import causing this error as when I comment it out the site renders correctly without error

**Comment by alexs-mparticle** [comment]
@mk-ryan1988 Sorry to hear you're having trouble. Would it be possible to include your package.json file and a brief snippet of how you're importing the SDK?

I haven't tested our SDK this on Vue but it should work with Webpack. We currently have a sample app built for React that uses Webpack and haven't seen these issues. Here's a link in case that offers some more insight: https://github.com/mParticle/mparticle-web-sample-apps/tree/main/core-sdk-samples/higgs-shop-sample-app

**Comment by mk-ryan1988** [comment]
Sure, I can's share the code as it private but I've setup a mock sandbox with the same package,json and example usage.
The sandbox is unlikely to compile due to some private dependancies being included

Let me know if you need anything else :)

**Comment by mk-ryan1988** [comment]
hey @alexs-mparticle did you get a chance to take a look at the sandbox? I doesn't seem to recreate the issue but I'm at a complete loss here

**Comment by alexs-mparticle** [comment]
@mk-ryan1988 I'm taking a look right now and just getting things set up. Nothing is jumping out at me right now, but I'll reach out if I have any further questions.

For now, please tell me what version of node you are using.

Also, can you try rolling back your version of mParticle you're importing? I want to rule out that maybe we recently introduced something that is colliding with something in your code base.

**Comment by mk-ryan1988** [comment]
We're currently on node 14 atm I'm afraid. Any version you'd recommend?

**Comment by alexs-mparticle** [comment]
I pulled your sample project locally and set it up based on the same version you are using and have had no issues building or running the code.

I am not as familiar with Vue as I am React, but I do wonder if there is a race condition caused between the time your application mounts vs when mParticle initializes.

In React, I've often used `useEffect` to initialize mParticle after the main app mounts. Not sure if there's something similar in Vue.

Perhaps you can also try wrapping `mParticleInit();` in a `setTimeout` in your `main.js`, like so:

```
setTimeout(function () {
  mParticleInit();
}, 100);
```

Lastly, try reaching out to your CSM/TSE to open a ticket if you need a faster support.

**Comment by mmustafa-tse** [comment]
Hi @mk-ryan1988,

Do you have an existing support ticket regarding this issue? If not, can you please file one using this [link](https://support.mparticle.com/)?

Thanks,

**Comment by mk-ryan1988** [comment]
Hey, I've not opened a support ticket yet as I went ahead an got a POC using the code snipped rather than self hosting. 

I did recently try installing an earlier version, I went back to 2.10.1, and this resolved the error. So there must be a breaking change in a later version

**Comment by mk-ryan1988** [comment]
Okay I did some further installing of various version and on our setup:

Project details:
node: 14.15.1
vue: 2.x

The error was introduced into version `2.16.2`.  
`2.16.1` loads without error but `2.16.2` and above all I replicate this issue

**Comment by alexs-mparticle** [comment]
Thanks for that feedback. I took a look at what changed between those versions and the main thing that jumps out at me is that we updated some things in our build process to more "modern" standards. Granted, that was over a year ago, and "modern" is relative.

I would be curious to know if the issue goes away using a more recent version of node. If memory serves, there were several projects were working with around that time that had issues on Node 14 and required some extra updates to get them to work on Node 15 and 16.
